docs/src/main/sphinx/admin/properties-general.rst

```diff
@@ -36,3 +36,22 @@ across nodes in the cluster. It can be disabled, when it is known that the
 output data set is not skewed, in order to avoid the overhead of hashing and
 redistributing all the data across the network. This can be specified
 on a per-query basis using the ``redistribute_writes`` session property.
+
+``protocol.v1.alternate-header-name``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**Type:** ``string``
+
+The 351 release of Trino changes the HTTP client protocol headers to start with
+``X-Trino-``. Clients for versions 350 and lower expect the HTTP headers to use
+``X-Presto-``, while newer clients expect ``X-Trino-``. You can support these
+older clients by setting this property to ``Presto``.
+
+The preferred approach to migrating from versions earlier than 351 is to update
+all clients together with the release, or immediately afterwards, and then
+remove usage of this property.
+
+Ensure to use this only as a temporary measure to assist in your migration
+efforts, for example if you continue to operate clusters with an older version
+of Trino.
```

Comment:
I feel the later half of the sentence might be confusing. May be removing this sentence, or just adding `Ensure to use this only as a temporary measure to assist in your migration efforts.` - might be better. I'd leave it to @mosabua.

Reply:
I am fine with cutting it short.